eessi/testsuite/hooks.py

```diff
@@ -105,6 +105,13 @@ def _assign_num_tasks_per_node(test: rfm.RegressionTest, num_per: int = 1):
     - num_tasks_per_node = num_per
     - num_cpus_per_task = test.default_num_cpus_per_node / num_tasks_per_node
     """
+
+    # OSU test inherits a default num_tasks_per_node = 1 from hpctestlib. There
+    # has to be a condition where a non-default value of num_per replaces the
+    # original num_tasks_per_node.
+    if test.num_tasks_per_node and num_per != 1:
+        if test.num_tasks_per_node != num_per:
+            test.num_tasks_per_node = num_per
```

Comment:
@this is not necessary and should be avoided. better to set it to None in the class body of the `class osu_pt_2_pt` (see suggestion)

Reply:
Agreed and the change has been incorporated.